### ISSUE TITLE
1.9 A quick introduction to HTML - Updating HTML examples.

### ISSUE THREAD

**Issue by stuartjnelson** [w3c/html#1426]
## Basic HTML example
```
<!DOCTYPE html>
<html>
  <head>
    <title>Sample page</title>
  </head>
  <body>
    <h1>Sample page</h1>
    <p>This is a <a href="demo.html">simple</a> sample.</p>
    <!-- this is a comment -->
  </body>
</html>
```
I think it would be good to make a few little changes to this so the example is more explicit and valid. I think updating;

1. Moving the link so that it's not nested inside the paragraph.
2. Making the text inside HTML elements/attributes describe what they are.

```
<!DOCTYPE html>
<html>
  <head>
    <title>Document title</title>
  </head>
  <body>
    <h1>Document heading</h1>
    <p>This is a paragraph element!</p>
    <a href="another-html-document.html">Link text for a link to another-html-document.html</a>.
    <!-- this is a comment -->
  </body>
</html>
```

## Nesting example
```
<!-- incorrect tag nesting -->
<p>This is <em>very <strong>wrong</em>!</strong></p>

<!-- appropriate tag nesting -->
<p>This <em>is <strong>correct</strong>.</em></p>
```

I think nesting is much clearer for beginners when it's shown indented on multiple lines. I also think it would be easier if the example wasn't using a `<p>` with phrasing content elements to avoid confusion with what can and can't be nested inside a `<p>`.

Maybe something along these lines using the elements that have already been introduced;
```
<!-- incorrect tag nesting -->
<h1>
    <a href="this-html-document.html">Document heading that is a link to this document
</h1>
    </a>

<!-- appropriate tag nesting -->
<h1>
    <a href="this-html-document.html">Document heading that is a link to this document</a>
</h1>
```

**Comment by LJWatson** [comment]
Thanks @stuartjnelson, appreciate the suggestions.

The only thing I'm not sure about, is moving the link outside of the paragraph (in your first example). Is it worth including two ```<p>``` elements in the example?
```
<p>This is a paragraph of text.</p>
<p><a href="...">Link text</a></p>
```

In any case, if you'd like to submit a PR with your suggested changes, you'd be very welcome!

**Comment by stuartjnelson** [comment]
@LJWatson thanks for the feedback! 

TLDR: Is it valid with HMLT 5.2 to have this anchor as a child of the paragraph?

I'll be honest I'm a little confused with HTML 5.2 what can and cannot be a child of a paragraph element. [MDN says anchors sometimes can be](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Phrasing_content) but only if it contains only phrasing content. I know my Vue.js linter recently has been kicking me whenever there is an anchor inside a paragraph.

My thoughts were to remove and ambiguity to not have an anchor inside the paragraph. On the other hand, link are a pretty fundamental part of an HTML document and it would be good for it to be there... Any clarity you can shed on this would be great because it's been bothering me for a few months now!

**Comment by scottaohara** [comment]
yes, @stuartjnelson `a` elements are allowed children of paragraphs. It's one of their primary use cases.

Under HTML5, the `a` element has a [transparent content model](http://w3c.github.io/html/dom.html#transparent). 
TLDR; For example, this means that while an `a` can wrap a `div`, it can not do so while also being the child of a `p`, because `div`s are invalid children of `p` elements.

For instance:
```
<!-- valid -->
<p>
  <a href="...">This is expected usage</a>. 
</p>

<!-- invalid -->
<p>
  <a href="...">
    <div>
        Not allowed because of the child div.
    </div>
  </a>
</p>
```

The invalid example is what the MDN content ([also noted in the HTML spec](http://w3c.github.io/html/dom.html#phrasing-content)) is referring to.

Regarding your linter, I assume it's because the `a` contains something like a `div` or other invalid content.  If your usage is like the valid example, then I'd submit there may be something incorrectly configured with the linter, or another error that's the true culprit here.

Let us know if you have any other questions :)

**Comment by stuartjnelson** [comment]
Hi @scottaohara. Thanks for clearing that up!! I did find it super weird that the linter was saying it was invalid... The linter was freaking out about articles I was pulling from Medium. I think I need to review my linting setup.

**Comment by scottaohara** [comment]
closed via #1436
